lib/appc.js

```diff
@@ -508,10 +508,59 @@ const Appc = {
 		});
 	},
 
+	/**
+	 * Run `appc whoami` command
+	 *@returns {Promise}
+		*/
+	getCurrentSession() {
+		return new Promise((resolve, reject) => {
+			let result = '';
+			this.testAppcCommand(success => {
+				if (success) {
+					new BufferedProcess({
+						command: atom.config.get('appcelerator-titanium.general.appcCommandPath'),
+						args: [ 'whoami', '-o', 'json' ],
+						stdout: data => result += data,
+						stderr: () => this.showAppcCommandError('You are not logged in.'),
+						exit: () => {
+							if (result && result.length) {
+								result = JSON.parse(result);
+								if (Object.entries(result).length === 0 && result.constructor === Object) {
+									reject();
+								}
+								resolve(result);
+							}
+						}
+					});
+				}
+			});
+		}).then(function (session) {
+			return new Promise((resolve, reject) => {
+				let result = '';
+				new BufferedProcess({
+					command: atom.config.get('appcelerator-titanium.general.appcCommandPath'),
+					args: [ 'config', '-o', 'json' ],
+					stdout: data => result += data,
+					exit: () => {
+						if (result && result.length) {
+							const config = JSON.parse(result);
+							if (Object.entries(config).length === 0 && config.constructor === Object) {
+								reject();
```

Comment:
return reject(); here, otherwise the execution will continue on and call resolve